src/main/java/org/jabref/logic/ai/chatting/model/Chat4AllModel.java

```diff
@@ -0,0 +1,187 @@
+                    .uri(URI.create(fullUrl))
+                    .header("Content-Type", "application/json")
+                    .POST(HttpRequest.BodyPublishers.ofString(requestBody))
+                    .timeout(Duration.ofSeconds(60))
+                    .build();
+
+            HttpResponse<String> response = httpClient.send(httpRequest, HttpResponse.BodyHandlers.ofString());
+            LOGGER.info("Chat4All response: {}", response.body());
+
+            TextGenerationResponse textGenerationResponse = gson.fromJson(response.body(), TextGenerationResponse.class);
+            if (textGenerationResponse.getChoices() == null || textGenerationResponse.getChoices().isEmpty()) {
+                throw new IllegalArgumentException("No choices returned in the response");
+            }
+
+            String generatedText = textGenerationResponse.getChoices().getFirst().getMessage().getContent();
+            if (generatedText == null || generatedText.isEmpty()) {
+                throw new IllegalArgumentException("Generated text is null or empty");
+            }
+            return new Response<>(new AiMessage(generatedText), new TokenUsage(0, 0), FinishReason.OTHER);
+        } catch (Exception e) {
+            LOGGER.error("Error generating message from Chat4All", e);
+            throw new RuntimeException("Failed to generate AI message", e);
+        }
+    }
+
+    private static class TextGenerationRequest {
+        protected final String model;
+        protected final List<Message> messages;
+        protected final Double temperature;
+        protected final Integer max_tokens;
+
+        private TextGenerationRequest(Builder builder) {
+            this.model = builder.model;
+            this.messages = builder.messages;
+            this.temperature = builder.temperature;
+            this.max_tokens = builder.max_tokens;
+        }
+
+        public static Builder builder() {
+            return new Builder();
+        }
+
+        public static class Builder {
+            private String model;
+            private List<Message> messages;
+            private Double temperature;
+            private Integer max_tokens;
+
+            public Builder model(String model) {
+                this.model = model;
+                return this;
+            }
+
+            public Builder messages(List<Message> messages) {
+                this.messages = messages;
+                return this;
+            }
+
+            public Builder temperature(Double temperature) {
+                this.temperature = temperature;
+                return this;
+            }
+
+            public Builder max_tokens(Integer max_tokens) {
+                this.max_tokens = max_tokens;
+                return this;
+            }
+
+            public TextGenerationRequest build() {
+                return new TextGenerationRequest(this);
+            }
+        }
+    }
+
+    private static class TextGenerationResponse {
```

Comment:
This and other classes can be converted to records 😅

Reply:
ok, I will change that later. If there are some other suggestions, please let me know.